sdk/python/kfp/components/types/type_utils.py

```diff
@@ -105,27 +105,28 @@ def is_parameter_type(type_name: Optional[Union[str, dict]]) -> bool:
 
 
 def get_artifact_type_schema(
-    artifact_class_or_type_name: Optional[Union[str,
-                                                Type[artifact_types.Artifact]]]
-) -> pipeline_spec_pb2.ArtifactTypeSchema:
+        schema_title: Union[str, dict],
+        schema_version: Union[str,
+                              None]) -> pipeline_spec_pb2.ArtifactTypeSchema:
     """Gets the IR I/O artifact type msg for the given ComponentSpec I/O
     type."""
-    artifact_class = artifact_types.Artifact
-    if isinstance(artifact_class_or_type_name, str):
-        if re.match(_GOOGLE_TYPES_PATTERN, artifact_class_or_type_name):
-            return pipeline_spec_pb2.ArtifactTypeSchema(
-                schema_title=artifact_class_or_type_name,
-                schema_version=_GOOGLE_TYPES_VERSION,
-            )
-        artifact_class = _ARTIFACT_CLASSES_MAPPING.get(
-            artifact_class_or_type_name.lower(), artifact_types.Artifact)
-    elif inspect.isclass(artifact_class_or_type_name) and issubclass(
-            artifact_class_or_type_name, artifact_types.Artifact):
-        artifact_class = artifact_class_or_type_name
+    # handles compatibility with v1 component YAML that use Google types (where schema_version is not included in YAML)
+    if re.match(_GOOGLE_TYPES_PATTERN, schema_title):
+        return pipeline_spec_pb2.ArtifactTypeSchema(
+            schema_title=schema_title,
+            schema_version=_GOOGLE_TYPES_VERSION,
+        )
+    if schema_title.lower() in _ARTIFACT_CLASSES_MAPPING:
+        # handles the way we currently truncate system.Artifact(s) to the suffix (Artifact, Dataset)
+        schema_title = _ARTIFACT_CLASSES_MAPPING.get(
+            schema_title.lower()).schema_title
+        # handles v1 artifacts that do not have a schema_version
+        schema_version = schema_version or DEFAULT_ARTIFACT_SCHEMA_VERSION
 
     return pipeline_spec_pb2.ArtifactTypeSchema(
```

Comment:
In the ideal state this is just a pass-through function that constructs an `ArtifactTypeSchema`, but there are special cases and backward compatibility cases we need to handle.